src/main/java/com/epam/grid/engine/provider/queue/slurm/SlurmQueueProvider.java

```diff
@@ -0,0 +1,370 @@
+
+        if (updateHostListDecrypted.equals(currentNodesDecrypted) && updateUserGroups.equals(userGroups)) {
+            throw new GridEngineException(HttpStatus.BAD_REQUEST, "New partition properties and the current one are "
+                    + "equal");
+        }
+        fillContextWithDataToUpdate(context, updateUserGroups, updateHostListDecrypted);
+
+        final CommandResult result = simpleCmdExecutor.execute(commandCompiler.compileCommand(getProviderType(),
+                SCONTROL_COMMAND, context));
+        checkIsResultIsCorrect(result);
+
+        final Map<String, Integer> slotDescription = getNodesDescription(updateHostListDecrypted,
+                CPUS_PER_NODE_BY_DEFAULT);
+
+        return Queue.builder()
+                .name(updateName)
+                .hostList(updateHostListDecrypted)
+                .allowedUserGroups(updateUserGroups)
+                .slots(new SlotsDescription(cpus, slotDescription))
+                .build();
+    }
+
+    @Override
+    public List<Queue> listQueues() {
+        final CommandResult result = simpleCmdExecutor.execute(commandCompiler.compileCommand(getProviderType(),
+                SINFO_COMMAND, new Context()));
+        checkIsResultIsCorrect(result);
+
+        return fillQueueNameFromOutput(result.getStdOut());
+    }
+
+    @Override
+    public List<Queue> listQueues(final QueueFilter queueFilter) {
+        final Context context = new Context();
+        if (queueFilter.getQueues() != null) {
+            context.setVariable(SCONTROL_PARTITION_NAME, queueFilter.getQueues());
+        }
+        final CommandResult result = simpleCmdExecutor.execute(commandCompiler.compileCommand(getProviderType(),
+                SINFO_COMMAND, context));
+        checkIsResultIsCorrect(result);
+
+        return fillQueueData(result.getStdOut());
+    }
+
+    @Override
+    public Queue deleteQueues(final String queueName) {
+        if (queueName == null) {
+            throw new GridEngineException(HttpStatus.BAD_REQUEST, "Partition name for deletion should be specified");
+        }
+        final Context context = new Context();
+        context.setVariable(SCONTROL_ACTION, SCONTROL_DELETE_COMMAND);
+        context.setVariable(SCONTROL_PARTITION_NAME, queueName);
+        final CommandResult result = simpleCmdExecutor.execute(commandCompiler.compileCommand(getProviderType(),
+                SCONTROL_COMMAND, context));
+        checkIsResultIsCorrect(result);
+        return Queue.builder()
+                .name(queueName)
+                .build();
+    }
+
+    private List<String> sortList(final List<String> list) {
+        return CollectionUtils.isEmpty(list)
+                ? list
+                : list.stream().sorted().collect(Collectors.toList());
+    }
+
+    private List<String> decryptGroupOfNodes(final List<String> hostList) {
+        final List<String> decryptedNodes = new ArrayList<>();
```

Comment:
I believe all this could be rewritten with stream API

Reply:
done